src/parser/comments.js

```diff
@@ -25,20 +25,20 @@
  */
 
 import Parser from "./index";
+import * as N from "../types";
 
-function last(stack) {
+function last<T>(stack: $ReadOnlyArray<T>): T {
   return stack[stack.length - 1];
 }
 
-const pp = Parser.prototype;
-
-pp.addComment = function (comment) {
+Parser.mixin(class extends Parser {
+addComment(comment: N.Comment): void {
```

Comment:
Some of them could be omitted, but I've tested around and some of them can't without breaking compilation. I also think it's nice to have the type annotations locally so I don't have to look them up in `parserTypes.js`. The code worked fine without *any* type annotations, the point was to make it easier to read.